segmentation_models_pytorch/encoders/timm_universal.py

```diff
@@ -23,34 +75,138 @@ def __init__(
             out_indices=tuple(range(depth)),
         )
 
-        # not all models support output stride argument, drop it by default
+        # Ｎot all models support output stride argument, drop it by default
         if output_stride == 32:
             common_kwargs.pop("output_stride")
 
-        self.model = timm.create_model(
-            name, **_merge_kwargs_no_duplicates(common_kwargs, kwargs)
-        )
+        # Load a temporary model to analyze its feature hierarchy
+        try:
+            with torch.device("meta"):
+                tmp_model = timm.create_model(name, features_only=True)
+        except Exception:
+            tmp_model = timm.create_model(name, features_only=True)
+
+        # Check if model output is in channel-last format (NHWC)
+        self._is_channel_last = getattr(tmp_model, "output_fmt", None) == "NHWC"
+
+        # Determine the model's downsampling pattern and set hierarchy flags
+        encoder_stage = len(tmp_model.feature_info.reduction())
+        reduction_scales = tmp_model.feature_info.reduction()
```

Comment:
To make it a bit more robust, it might be a tuple

```suggestion
        reduction_scales = list(tmp_model.feature_info.reduction())
```